### PR TITLE
closes #1443

### PR DESCRIPTION
**Pull request by aMarCruz**
Cleanup trigger and fix #1443

### CODE REVIEW

**Comment by aMarCruz** [comment]
I like the `error` event. I can use it on tmpl:

``` js
if (context.trigger) context.trigger('error', e)
else if (errorHandler) ...
```

`tmpl.errorHandler` at app level, `error` event for granular control.

**Comment by GianlucaGuarini** [comment]
@aMarCruz I have removed the `try/catch` block from the sourcecode I think we can handle the `tmpl.errorHandler` directly in riot I will merge the other stuff I like from this pull request ok?

**Comment by aMarCruz** [comment]
@GianlucaGuarini ok, and it could be a breaking change, so `tmpl` will keep the current behavior for errors out of expressions.

File: dist/es6.observable.js [hunk 1]
```diff
@@ -87,15 +88,11 @@ var observable = function(el) {
 
     // getting the arguments
     // skipping the first one
-    var arglen = arguments.length - 1,
```
Comment (mobxu) [inline dist/es6.observable.js]: Vsglie

File: test/specs/core.specs.js [hunk 2]
```diff
@@ -363,5 +363,30 @@ describe('Core specs', function() {
...
+
+    el.on('e', function () { throw new Error('foo') })
+      .one('error', function (e) { err = e.message })
+      .trigger('e')
+    expect(err).to.contain('foo')
+
+    // expect(el.trigger).withArgs('e').to.throw() //can I make work this?
+    try { el.trigger('e') }
+    catch (e) { counter = -1 }
+    expect(counter).to.be(-1)
```
Comment (mobxu) [inline test/specs/core.specs.js]: HuanuXpceo

**Pull request closed without merging**